src/frontend/ocamlmerlin/query_json.ml

```diff
@@ -112,6 +112,14 @@ let dump (type a) : a t -> json =
         );
       "position", mk_position pos;
     ]
+  | Syntax_document (identifier, pos) ->
+    mk "syntax-document" [
+      "identifier", (match identifier with
```

Comment:
Is there a specific reason why we need an identifier to perform the query ?

Reply:
I understand an identifier to be like, a user selected/highlighted some code, if this is the wrong idea then I don't think we need an identifier to run the query and will probably remove it alltogether

Reply:
I think for the first iteration we can restrict ourselves to the position of the cursor and the node(s) around that 👍